api/oauth.go

```diff
@@ -47,6 +49,14 @@ func registerOAuthApp(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if *utils.Cfg.ServiceSettings.EnableOnlyAdminIntegrations {
+		if !(c.IsSystemAdmin() || c.IsTeamAdmin()) {
```

Comment:
`IsTeamAdmin` will return true if the user is a system admin, so you can probably get rid of `c.IsSystemAdmin()` here